src/components/CoinDisplay.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <span class="d-flex">
+    <template v-for="(value, key) in splitValues(value)" :key="key">
+      <img 
+        v-if="(value != '')" 
+        :src="getCoinImagePath(key)" 
+        class="img-fuid h-100 align-self-center mx-1" 
+      />
+      {{ value }}
+    </template>
+  </span>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+type Coin = {
+  bronze: string;
+  silver: string;
+  gold: string;
+  plat: string;
+  dem: string;
+};
+
+export default defineComponent({
+  name: "CoinDisplay",
+  props: {
+    value: {
+      required: true,
+      type: String,
+    },
+  },
+  methods: {
+    splitValues(value: string): Coin {
+      let coins = {
+        dem: "",
+        plat: "",
+        gold: "",
+        silver: "",
+        bronze: "",
+      };
+
+      coins["bronze"] = value.slice(-2);
+      value = value.slice(0, -2);
+      coins["silver"] = value.slice(-2);
+      value = value.slice(0, -2);
+      coins["gold"] = value.slice(-2);
+      value = value.slice(0, -2);
+      coins["plat"] = value.slice(-2);
+      value = value.slice(0, -2);
+      coins["dem"] = value.slice(-2);
+      value = value.slice(0, -2);
+
+      return coins;
+    },
```

Comment:
Maybe i'm mistaken here but are you trying to cut a coin into its different value parts?

Take a number like 12345 coins:
```typescript
let v = 12345;
let coins = [];
let n = 5; // number of distinct coin types
while (n--) {
  let c = Math.pow(100, n)
  coins.push(Math.floor(v / c))
  v %= c
}

// coins is now [0, 0, 1, 23, 45]
```

This way you don't need to have a dictionary storing each part of the coin. If you ever need the specific number of a certain coin type, you can always get it by index e.g. platinum is `coins[1]`.